DataModel/SubSample.h

```diff
@@ -3,27 +3,48 @@
 
 #include <iostream>
 #include <vector>
+#include <stdint.h>
 
 class SubSample{
 
  public:
 
-  SubSample() {};
+  /// Type for relative hit times within a SubSample. Unit = ns
+  typedef float relative_time_t;
 
-  SubSample(std::vector<int> PMTid,std::vector<float> time);
+  /// Type for absolute timestamps of a SubSample. Unit = us
+  typedef int64_t timestamp_t;
 
-  SubSample(std::vector<int> PMTid, std::vector<float> time, std::vector<float> charge);
+  /// Factor between unit of timestamp and hit times
+  static const relative_time_t s_TIMESTAMP_TO_RELATIVE_TIME = 1e3; //  = 1 us/ns
 
-  std::vector<int> m_PMTid;
-  std::vector<float> m_time;
-  std::vector<float> m_charge;
-  std::vector<int> m_time_int;
-  std::vector<int> m_charge_int;
+  SubSample() {};
+  SubSample(std::vector<int> PMTid, std::vector<relative_time_t> time);
+  SubSample(std::vector<int> PMTid, std::vector<relative_time_t> time, std::vector<float> charge);
+  SubSample(std::vector<int> PMTid, std::vector<relative_time_t> time, std::vector<float> charge, timestamp_t timestamp);
```

Comment:
Do we need 2 versions of this? This will simplify:
```suggestion
  SubSample(std::vector<int> PMTid, std::vector<relative_time_t> time, std::vector<float> charge, timestamp_t timestamp = 0);
```

Reply:
My thinking was that I want to keep the timestamp left of the hit time, because in general it should be the thing with the larger unit (us instead of ns). Also I have read that there are some instances where it is preferable to use overloading over default values: https://google.github.io/styleguide/cppguide.html#Default_Arguments

I am not wedded to this decision, though.

Reply:
I'm not sure I understand what "timestamp left of the hit time" means, because in the constructor with both, the timestamp is the last argument (i.e. right-most)
Anyway, instead of default constructor arguments, we could instead do constructors calling constructors in the implementation to achieve the same thing, and to reduce code repetition https://en.wikipedia.org/wiki/C++11#Object_construction_improvement

Reply:
Ahh, sorry. I did not pay attention and was confusing the constructor and the `TimeDifference` method here. Please ignore the argument order argument.

I tried to use the "constructor calls constructor" method, but my compiler did not understand that. I think the default compiler on CentOS7 does not have full C++11 support. Apparently there is a flag to enable experimental support (from before the standard was standardised), but I was not sure I should be using that (and add the flag to the makefile).

Reply:
Maybe make a method call Construct() that all constructors call? A decision on C++11/14/17/20 support is probably for another PR